src/dags/importscripts/import_processenverbaalverkiezingen.py

```diff
@@ -0,0 +1,134 @@
+            self.connection.retrlines("LIST", lambda x: list_.append(x.split()))
+            for info in list_:
+                type, name = info[0], info[-1]
+                if type.startswith("d"):
+                    dirs.append(name)
+                else:
+                    files.append(name)
+            return dirs, files
+
+    def traverse_folder(self, path: str = "/") -> Generator:
+        """Recursive walk through directory tree, based on a BFS algorithm.
+        This function acts like an orchestrator for looking for files and dirs.
+
+        Args:
+            path: The path for looking for files and directories at the same level
+
+        Yields:
+            list of all files incl its path until all directories are depleted
+
+        """
+        dirs, files = self.list_dirs_and_files(path)
+        yield path, dirs, files
+        for name in dirs:
+            path = ospath.join(path, name)
+            yield from self.traverse_folder(path)
+            self.connection.cwd("..")
+            path = ospath.dirname(path)
+
+
+def save_data(
+    startfolder: str, prefix_url: str, host: str, user: str, passwd: str, output_file: str
+) -> None:
+    """Save listing of data files to csv
+
+    Args:
+        startfolder: The starting directory to start looking for files and directories
+        prefix_url: The protocol, subdomain and domain part of the URI to locate files
+        host: the hostname of the objectstore where files are located
+        user: the username that can access the objectstore
+        passwd: the password that is used to access the objectstore
+        output_file: name of .csv file to save
+
+    Executes:
+        Stores file URL's and it's metadata to .csv file
+
+    Notes:
+        The filenames are meaningful. It contains it's metadata. For example:
+        `001.procesverbaaltk21.Amstel1.pdf` conceals <volgnummer>.<documentnaam>.<stemlocatie>.pdf
+        Furthermore, the files are located in a folder which name
+        represents it's election year i.e. 2021
+
+    """
+    data_to_save: List = []
+    connection = ftplib.FTP(host=host)
+    connection.login(user=user, passwd=passwd)
+    get_listing = FTPListing(connection)
+
+    for data in get_listing.traverse_folder(startfolder):
+        for file in data[2]:
+
+            verkiezingsjaar = data[0].split("/")[1]
+            volgnummer = file.split(".")[0]
+            uri = URL(prefix_url) / data[0] / file
+            documentnaam = file.split(".")[1]
+            stemlocatie = file.split(".")[2]
+
+            metadata = Data(
+                verkiezingsjaar=verkiezingsjaar,
+                volgnummer=volgnummer,
+                uri=uri,
+                documentnaam=documentnaam,
+                stemlocatie=stemlocatie,
+                id=verkiezingsjaar + volgnummer,
+            )
+            data_to_save.append(metadata)
+
+    header = Data.__annotations__.keys()
+    data = [row.values() for row in data_to_save]
```

Comment:
You're doing what [csv.DictWriter](https://docs.python.org/3.8/library/csv.html?highlight=dictwriter#csv.DictWriter) does for you automatically.

In other words. Use `csv.DictWriter` and you can write each dict directly (in a loop).

Reply:
Should you choose to stick to your current implement than at least use `newline=""` in your call to `open`.

See also: https://docs.python.org/3.8/library/csv.html?highlight=dictwriter#id3 (the footnote of that document)

Reply:
I like the DictWriter approach more, I adopted your suggestion